lib/pacemaker/pcmk_sched_allocate.c

```diff
@@ -740,6 +746,10 @@ unpack_cib(xmlNode *cib, unsigned long long flags, pe_working_set_t *data_set)
      */
     set_working_set_defaults(data_set);
 
+    if (localhost_save) {
+        data_set->localhost = localhost_save;
+    }
+
```

Comment:
Was surprised as well when I had checked earlier today ...
Anyway for the case it really matters wouldn't it make sense to explicitly pass the local-node-name in
via a parameter for the case it is called via stonith-device-update? Anyway solely private api - right ...

Reply:
> To avoid effects on any other code, I'm thinking we should re-add the cluster_status() call before pcmk__schedule_actions() in cib_devices_update() instead of changing update_cib().

I think the changes will be as follows, but what about?

```
(snip)
    fenced_data_set->input = local_cib;
    fenced_data_set->localhost = stonith_our_uname;
    pe__set_working_set_flags(fenced_data_set, data_set_flags);
    cluster_status(fenced_data_set);
    pcmk__schedule_actions(local_cib, data_set_flags, fenced_data_set);
(snip)
```

Reply:
Yes, that should do it

Reply:
Actually, looking at it again, I like your original version better. Nothing besides the fencer sets localhost, so it can't affect anything else

Reply:
HI Ken,

All right!

Thansk!

Reply:
@HideoYamauchi , 

I still don't understand why it causes problems, though. The fencer needs to set localhost in standalone mode (used for the regression tests), since there is no CIB, but I would expect the local node should otherwise always be in the CIB. Did you figure out how this fixes your issue?

Reply:
HI Ken,

I have issued a new revised PR, but I will investigate it a little more.

Reply:
Hi Ken,

First, the problem does not occur if /var/lib/pacemaker/cib/cib.xml already exists.
Occurs when building a new cluster and pouring cib.xml with pcs cluster cib-push.
Also, if the node is set in the nodes section in the cib.xml to be poured, the problem does not occur.


Initially the node of local_cib is set correctly, but If there is no node section in the flowing cib.xml, the node of local_cib will be deleted after applying the difference of cib of fenced with xml_apply_patchset().
For this reason, the node information cannot be found from local_cib.

```
static voidupdate_cib_cache_cb(const char *event, xmlNode * msg){
(snip)
crm_log_xml_info(local_cib, "### YAMAUCHI PREV :");
        patchset = get_message_xml(msg, F_CIB_UPDATE_RESULT);
        xml_log_patchset(LOG_TRACE, "Config update", patchset);
        rc = xml_apply_patchset(local_cib, patchset, TRUE);
crm_log_xml_info(local_cib, "### YAMAUCHI AFT :");
```

```
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :   <cib crm_feature_set="3.15.1" validate-with="pacemaker-3.8" epoch="1" num_updates="0" admin_epoch="0">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :     <configuration>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :       <crm_config/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :       <nodes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :         <node id="1" uname="rh85-01sc"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :         <node id="2" uname="rh85-02sc"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :         <node id="3" uname="rh85-03sc"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :       </nodes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :       <resources/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :       <constraints/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :     </configuration>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :     <status/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI PREV :   </cib>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :   <cib crm_feature_set="3.11.0" validate-with="pacemaker-3.7" epoch="24" num_updates="0" admin_epoch="0">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :     <configuration>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       <crm_config>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <cluster_property_set id="cib-bootstrap-options">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <nvpair id="cib-bootstrap-options-pe-input-series-max" name="pe-input-series-max" value="0"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <nvpair id="cib-bootstrap-options-pe-warn-series-max" name="pe-warn-series-max" value="0"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <nvpair id="cib-bootstrap-options-pe-error-series-max" name="pe-error-series-max" value="0"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <nvpair id="cib-bootstrap-options-no-quorum-policy" name="no-quorum-policy" value="freeze"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </cluster_property_set>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       </crm_config>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       <nodes/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       <resources>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <group id="fence1-group">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="stonith" id="fence1-vmware-rest" type="fence_vmware_rest">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <instance_attributes id="fence1-vmware-rest-instance_attributes">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-ip" name="ip" value="192.168.28.250"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-password" name="password" value="nttdata01@A"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-pcmk_host_map" name="pcmk_host_map" value="rh85-01sc:rh85-01sc"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-pcmk_reboot_action" name="pcmk_reboot_action" value="reboot"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-ssl_insecure" name="ssl_insecure" value="1"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-username" name="username" value="administrator@vsphere.local"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence1-vmware-rest-instance_attributes-disable_timeout" name="disable_timeout" value="false"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </instance_attributes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence1-vmware-rest-monitor-interval-3600s" interval="3600s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence1-vmware-rest-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence1-vmware-rest-stop-interval-0s" interval="0s" name="stop" on-fail="ignore" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </group>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <group id="fence2-group">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="stonith" id="fence2-vmware-rest" type="fence_vmware_rest">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <instance_attributes id="fence2-vmware-rest-instance_attributes">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-ip" name="ip" value="192.168.28.250"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-password" name="password" value="nttdata01@A"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-pcmk_host_map" name="pcmk_host_map" value="rh85-02sc:rh85-02sc"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-pcmk_reboot_action" name="pcmk_reboot_action" value="reboot"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-ssl_insecure" name="ssl_insecure" value="1"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-username" name="username" value="administrator@vsphere.local"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence2-vmware-rest-instance_attributes-disable_timeout" name="disable_timeout" value="false"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </instance_attributes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence2-vmware-rest-monitor-interval-3600s" interval="3600s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence2-vmware-rest-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence2-vmware-rest-stop-interval-0s" interval="0s" name="stop" on-fail="ignore" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </group>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <group id="fence3-group">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="stonith" id="fence3-vmware-rest" type="fence_vmware_rest">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <instance_attributes id="fence3-vmware-rest-instance_attributes">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-ip" name="ip" value="192.168.28.250"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-password" name="password" value="nttdata01@A"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-pcmk_host_map" name="pcmk_host_map" value="rh85-03sc:rh85-03sc"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-pcmk_reboot_action" name="pcmk_reboot_action" value="reboot"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-ssl_insecure" name="ssl_insecure" value="1"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-username" name="username" value="administrator@vsphere.local"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="fence3-vmware-rest-instance_attributes-disable_timeout" name="disable_timeout" value="false"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </instance_attributes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence3-vmware-rest-monitor-interval-3600s" interval="3600s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence3-vmware-rest-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="fence3-vmware-rest-stop-interval-0s" interval="0s" name="stop" on-fail="ignore" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </group>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <group id="pgsql-group">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="ocf" id="filesystem1" provider="heartbeat" type="Dummy">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem1-monitor-interval-10s" interval="10s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem1-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem1-stop-interval-0s" interval="0s" name="stop" on-fail="fence" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="ocf" id="filesystem2" provider="heartbeat" type="Dummy">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem2-monitor-interval-10s" interval="10s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem2-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem2-stop-interval-0s" interval="0s" name="stop" on-fail="fence" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="ocf" id="filesystem3" provider="heartbeat" type="Dummy">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem3-monitor-interval-10s" interval="10s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem3-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="filesystem3-stop-interval-0s" interval="0s" name="stop" on-fail="fence" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="ocf" id="ipaddr" provider="heartbeat" type="Dummy">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="ipaddr-monitor-interval-10s" interval="10s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="ipaddr-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="ipaddr-stop-interval-0s" interval="0s" name="stop" on-fail="fence" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="ocf" id="pgsql" provider="heartbeat" type="Dummy">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="pgsql-monitor-interval-10s" interval="10s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="pgsql-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="300s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="pgsql-stop-interval-0s" interval="0s" name="stop" on-fail="fence" timeout="300s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </group>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <clone id="ping-clone">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <primitive class="ocf" id="ping" provider="pacemaker" type="ping">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <instance_attributes id="ping-instance_attributes">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="ping-instance_attributes-attempts" name="attempts" value="2"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="ping-instance_attributes-host_list" name="host_list" value="192.168.28.254"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="ping-instance_attributes-multiplier" name="multiplier" value="1"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="ping-instance_attributes-name" name="name" value="ping-status"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <nvpair id="ping-instance_attributes-timeout" name="timeout" value="2"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </instance_attributes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="ping-monitor-interval-10s" interval="10s" name="monitor" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="ping-start-interval-0s" interval="0s" name="start" on-fail="restart" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :               <op id="ping-stop-interval-0s" interval="0s" name="stop" on-fail="ignore" timeout="60s"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             </operations>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </primitive>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </clone>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       </resources>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       <constraints>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_location id="location-fence1-group-rh85-01sc--INFINITY" node="rh85-01sc" rsc="fence1-group" score="-INFINITY"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_location id="location-fence2-group-rh85-02sc--INFINITY" node="rh85-02sc" rsc="fence2-group" score="-INFINITY"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_location id="location-fence3-group-rh85-02sc--INFINITY" node="rh85-03sc" rsc="fence2-group" score="-INFINITY"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_location id="location-pgsql-group-rh85-01sc-200" node="rh85-01sc" rsc="pgsql-group" score="200"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_location id="location-pgsql-group-rh85-02sc-100" node="rh85-02sc" rsc="pgsql-group" score="100"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_location id="location-pgsql-group" rsc="pgsql-group">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <rule boolean-op="or" id="location-pgsql-group-rule" score="-INFINITY">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <expression attribute="ping-status" id="location-pgsql-group-rule-expr" operation="lt" value="1"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :             <expression attribute="ping-status" id="location-pgsql-group-rule-expr-1" operation="not_defined"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           </rule>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </rsc_location>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_order first="ping-clone" first-action="start" id="order-ping-clone-pgsql-group-mandatory" symmetrical="false" then="pgsql-group" then-action="start"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <rsc_colocation id="colocation-pgsql-group-ping-clone-INFINITY" rsc="pgsql-group" score="INFINITY" with-rsc="ping-clone"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       </constraints>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       <rsc_defaults>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         <meta_attributes id="rsc_defaults-meta_attributes">
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <nvpair id="rsc_defaults-meta_attributes-migration-threshold" name="migration-threshold" value="1"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :           <nvpair id="rsc_defaults-meta_attributes-resource-stickiness" name="resource-stickiness" value="200"/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :         </meta_attributes>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :       </rsc_defaults>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :     </configuration>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :     <status/>
Jul 13 09:11:47 rh85-01sc pacemaker-fenced    [789731] (update_cib_cache_cb) 	info: ### YAMAUCHI AFT :   </cib>

```

Considering such an update in cib.xml, cib_devices_update() seems to set localhost.


Best Regards,
Hideo Yamauchi.

Reply:
Hi Ken,

> Actually, looking at it again, I like your original version better. Nothing besides the fencer sets localhost, so it can't affect anything else

OKay!

Many thanks,
Hideo Yamauchi.

Reply:
Is this the only issue we are seeing with such a node-section-purge?

Reply:
> Is this the only issue we are seeing with such a node-section-purge?

There are various patterns in the input of cib.xml, so I think I hit it this time.
Especially we often use cib.xml without node section.
I haven't checked if there are any other issues that may cause problems.

Many thanks,
Hideo Yamauchi.